### ISSUE TITLE
Prevent automerge failures

### ISSUE THREAD

**Issue by simoneb** [fastify/github-action-merge-dependabot#145]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

This was originally reported in a GitHub support ticket:

> There is an action which allows to merge dependabot Pull Requests automatically: https://github.com/fastify/github-action-merge-dependabot
>
> It can be used in workflows as this one: https://github.com/simoneb/jetl/blob/master/.github/workflows/ci.yml.
>
> You will see that it's used in a job automerge which needs job build. Job build is a required status check in order for the PR to be merged, configured in the branch protection rules of the repository settings.
>
> What I see happening sometimes is that the automerge job fails because build has not completed yet, hence the required status check is not satisfied and hence the PR cannot be merged. But because automerge depends on the successful completion of build it doesn't make sense that this happens. It happens intermittently and you can see one such case here: https://github.com/simoneb/jetl/runs/4899793357?check_suite_focus=true

GitHub's response:

> Hi Simone,
> 
> Thank you for reaching out to GitHub Support!
> 
> Taking a look at your workflow file, I wonder if the race condition comes from:
> 
> on: [push, pull_request]
> 
> I tested this out and when I update or create a new pull request, it also triggers the push event, as I'm technically pushing a pull request into the repository.
> 
> So I imagine there might be a race condition as the workflow is triggered twice when creating/updating the pull request and waiting on build.
> 
> With that being said, would it be possible to just use pull_request, or alternatively, we have had other users use workflow_run instead of needs, but the caveat to that is that thus event will only trigger a workflow run if the workflow file is on the default branch:
> 
> https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run
> 
> https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow
> 
> We hope this helps! If you have any other questions about using GitHub Actions, we recommend asking in our GitHub Community Forum:
> 
> https://github.community/c/code-to-cloud/52

**Comment by ramonmulia** [comment]
@simoneb After after diving into the issue I wasn't able to reproduce it.

It could be a race condition but no reason to skip the `build` step `Required status check "build" is in progress.`

The `pull_request` event triggers when a PR event occurs targeting the branches specified in the workflow and
`push` event is triggered whenever a commit is pushed to the specified branches.

Since a PR merge is just a commit pushed to the target branch it will trigger push events, 
I believe it could listen just `push` events instead of both (pull_request and push), as this action will get just `dependabot` PRs to process skipping those that aren't.

what do you think?

**Comment by simoneb** [comment]
Hi Ramon, I believe the solution is to simply restrict which push and pull_request events trigger the workflow.

Usually we want a CI workflow to run:

- on pull requests: whenever a PR is opened or new commits are pushed to it. I think this required the pull_request event
- on pushes to master: this requires the push event, but it can be restricted to only pushes to master

**Comment by ramonmulia** [comment]
Hey @simoneb, after some tests, I can point out the following scenarios:

1)

```
 on:   
     pull_request:
        types: [opened, edited]
    push: 
	branches:
		- ‘master’
```
		

-  It will trigger pull_request while creating a new PR, but if you commit something afterwards it will not dispatch the action again.
- The `edited` type is when you edit something into PR like description not when you commit something.

This way it will run only the first time you create the PR (and when you edit something into it) but NOT for new PR commits and also when it is merged to master.

2) 
```
on:
  push: 
```

It will work in all scenarios.
- When you create a new PR
- When you commit in the PR
- When merged to master or any branch

I mean, we don’t need to specify `pull_request` if we want this runs when the PR is opened.
